### DISCUSSION TITLE
Injecting both DV and HDR10Plus into one stream.

### DISCUSSION THREAD

**Discussion by vmiyoshi** [quietvoid/hdr10plus_tool#69] [Q&A]
So... I have extracted both DV, and HDR10plus from my source (which is a UHD made into an mkv using MakeMKV)
I converted the DV from dvhe.07.06, to dvhe.0806
Then I injected both DV, and HDR10Plus into my re-encoded .hevc
Here is the mediainfo - 
_HDR format: Dolby Vision, Version 1.0, dvhe.08.06, BL+RPU, HDR10 compatible / SMPTE ST 2094 App 4, Version 1, HDR10+ Profile B compatible_
Problem is the file only plays as HDR10 on my Samsung TV. Files originally sourced from stream services, which mediainfo looks the same as above, play fine as HDR10plus. 
Also if I only inject the HDR10plus metadata the file plays as HDR10plus, it's just that when I inject both types of metadata does it NOT play as plus. I already tried injecting DV RPU first, then the HDR10plus and Vice versa, however it seems like it doesn't make a differnce. 
I'm wondering what the "--drop-hdr10plus" flag means in dovi_tool? because I'm not using that when I'm extracting the DV from the source file. Should I be using this for sources that have both types of metadata, if I plan on extracting HDR10Plus using the HDR10plus_tool? I'm sorry I'm kind of confused...lol

**Comment by quietvoid** [comment]
> I'm wondering what the "--drop-hdr10plus" flag means in dovi_tool?

It removes HDR10+ so you probably wouldn't want to use it.
I don't know why your TV doesn't play videos with both, but there's nothing to be done about it here.

**Reply by vmiyoshi** [reply]
The Samsung TV does play files with both, just not files I create using these tools.
dovi_tool
HDR10plus_tool
mkvmerge

**Reply by quietvoid** [reply]
You said
> Also if I only inject the HDR10plus metadata the file plays as HDR10plus, it's just that when I inject both types of metadata does it NOT play as plus.

So clearly it doesn't correctly play files with both.

**Reply by vmiyoshi** [reply]
HDR format: _Dolby Vision, Version 1.0, dvhe.08.06, BL+RPU, HDR10 compatible / SMPTE ST 2094 App 4, Version 1, HDR10+ Profile B compatible_
Files originally sourced from stream services, with mediainfo that looks the same as above, play fine as HDR10plus.

**Reply by quietvoid** [reply]
Well I don't know but since HDR10+ alone works fine, it should work with Dolby Vision too.

**Reply by quietvoid** [reply]
Maybe if you had small samples of a working and not working file, then I could take a look.

**Reply by vmiyoshi** [reply]
Added Sample.zip 
The working sample = HG_100f.mkv
[Sample.zip](https://github.com/quietvoid/hdr10plus_tool/files/13806289/Sample.zip)

Non-working sample = DS_100f.mkv

**Reply by quietvoid** [reply]
There's no difference for HDR10+ between the files.
Your non-working sample is missing HDR10 metadata though, so it might be causing the problem.

**Reply by vmiyoshi** [reply]
I just want to say thank you for making these tools, and for taking the time to look into this for a stranger on the internet. 
I knew before your reply that the video stream was missing some of the metadata from the original source. However reading your reply sent me down a metadata rabbit hole. I found that both ffmpeg, and MKVExtract appear to delete this data, as I see the metadata in the mkv file, however it's not present in the hevc that's extracted. All this could be normal, this is beyond my level of understanding. What I found is there is an option in handbrake to keep metadata from source. MKVToolNix GUI shows this metadata  in the hevc steam which you can edit. I was already coping that data and pasting into the hevc stream that I added the dv/hrd10+ metadata to from your tools, all this in MKVToolNix GUI which is the tool I'm obviously using to do the remux. The outcome is "SMPTE ST 2086" is added to the "HDR format" string when looking a the file in mediainfo. Doing this didn't seem to affect the playback so I thought it was benign. Where it gets odd is MKVToolNix does not show the metadata from the source mkv file, only the one handbrake created (re-encoded from the source). Mediainfo does however show that metadata. Based on my research I believe this has to do with how the metadata is being presented in the file. I guess handbrake is presenting it in a format MKVToolNix understands. Anyway this metadata I added from what I saw in MKVToolNix in one stream, again doesn't change anything. If the file only has HDR10+ metadata adding using your tools, my TV shows the "HDR10+" with or without that added metadata. I'm using a 2022 Apple TV, the application playing the mkv files is infused (the official Plex app sucks, and doesn't pass HDR10+ but I digress). I bought a "zidoo Z9X PRO" off Amazon today, once it comes in I will re-test.  Again thank you, as time goes on we learn more and more about these formats and how applications, and displays handle them. I hate this new stage in the format wars...lol.
